app/1.3.x/getting-started/configuring-a-grpc-service.md

```diff
@@ -0,0 +1,283 @@
+---
+title: Configuring a gRPC Service
+---
+
+<div class="alert alert-warning">
+  <strong>Before you start:</strong>
+  <ol>
+    <li>Make sure you've <a href="https://konghq.com/install/">installed Kong</a> &mdash; It should only take a minute!</li>
+    <li>Make sure you've <a href="/{{page.kong_version}}/getting-started/quickstart">started Kong</a>.</li>
+  </ol>
+</div>
+
+Note: this guide assumes familiarity with gRPC; for learning how to set up
+Kong with an upstream REST API, check out the [Configuring a Service guide][conf-service].
+
+Starting with version 1.3, gRPC proxying is natively supported in Kong. In this
+section, you'll learn how to configure Kong to manage your gRPC services. For the
+purpose of this guide, we'll use [grpcurl][grpcurl] and [grpcbin][grpcbin] - they
+provide a gRPC client and gRPC services, respectively.
+
+We will describe two setups: Single gRPC Service and Route and single gRPC Service
+with multiple Routes. In the former, a single catch-all Route is configured, which
+proxies all matching gRPC traffic to an upstream gRPC service; the latter demonstrates
+how to use a Route per gRPC method.
+
+In Kong 1.3, gRPC support assumes gRPC over HTTP/2 framing. As such, make sure
+you have at least one HTTP/2 proxy listener (check out the [Configuration Reference][configuration-rerefence]
+for how to). In this guide, we will assume Kong is listening for HTTP/2 proxy
+requests on port 9080.
+
```

Comment:
Is it possible to add right startup commands for `kong` and `grpcbin`? Is there a reason that this guide uses different ports than the defaults? Defaults for Kong being `8000` and `8443` with `http2` flag added /which we should add by default as of now), and default for `grpcbin` being `9000` and `9001` (`docker run -it --rm -p 9000:9000 -p 9001:9001 moul/grpcbin`).

Reply:
Given HTTP/1.1 and HTTP/2 are different protocols, and client compatibility is not transparent, IMO it makes sense to use a separate port, so as to ensure examples in this page are clearly distinguishable from those in other pages (where HTTP/2 isn't assumed/used).

As for grpcbin, ports 9000 and 9001 are close to the default ports in Kong -- and in fact used by default in Kong, e.g., in tests, so choosing different ports avoids conflicts.

Reply:
Ok, it was just an suggestion. Perhaps then add right start commands as an example.

Just leaving this here (I am sure you are aware of that):
the `http2` flag just sets so that port can handle http/2 too, in addition to http/1.1. In most realworld deployments they are run at the same time on same port and upgraded as needed.